src/bootstrap/src/utils/cc_detect.rs

```diff
@@ -223,6 +223,16 @@ fn default_compiler(
             }
         }
 
+        t if t.contains("-wasi") => {
+            let root = PathBuf::from(std::env::var_os("WASI_SDK_PATH")?);
+            let compiler = match compiler {
+                Language::C => format!("{t}-clang"),
+                Language::CPlusPlus => format!("{t}-clang++"),
```

Comment:
Out of interest, why is clang/clang++ prefixed with the target? That seems like an odd choice for the usually cross-compiling nature of clang/LLVM?

(Obviously not blocking here since it's not something we can change on the Rust side :)

Reply:
It's not necessarily required but it helps configure clang with the `--target` value. Clang has a feature that a target-prefixed executable doesn't need `--target` passed, and the wasi-sdk sysroot has a bunch of symlinks for wasi targets that all point at a single `clang` executable, so it's all the same executable just a name-based-dispatch executable.

Reply:
Ah I didn't know about that feature. Sounds good.